src/utils/process.rs

```diff
@@ -1,32 +1,108 @@
 use std::collections::{HashMap, HashSet};
 use sysinfo::{Pid, Process, ProcessExt, SystemExt};
 
+// Return value of `extract_args(args: &[String]) -> ProcessArgs<T>` function which is
+// passed to `calling_process_cmdline()`.
 #[derive(Debug, PartialEq)]
-pub enum GitBlameExtension {
-    Some(String),
-    None,
-    NotGitBlame,
+pub enum ProcessArgs<T> {
+    // A result has been successfully extracted from args.
+    Args(T),
+    // The extraction has failed.
+    ArgError,
+    // The process does not match, others may be inspected.
+    OtherProcess,
 }
 
 pub fn git_blame_filename_extension() -> Option<String> {
+    calling_process_cmdline(blame::guess_git_blame_filename_extension)
+}
+
+mod blame {
+    use super::*;
+
+    // Skip all arguments starting with '-' from `args_it`. Also skip all arguments listed in
+    // `skip_this_plus_parameter` plus their respective next argument.
+    // Keep all arguments once a '--' is encountered.
+    // (Note that some arguments work with and without '=': '--foo' 'bar' / '--foo=bar')
+    fn skip_uninteresting_args<'a, 'b, ArgsI, SkipI>(
+        mut args_it: ArgsI,
+        skip_this_plus_parameter: SkipI,
+    ) -> Vec<&'a str>
+    where
+        ArgsI: Iterator<Item = &'a str>,
+        SkipI: Iterator<Item = &'b str>,
+    {
+        let arg_follows_space: HashSet<&'b str> = skip_this_plus_parameter.into_iter().collect();
+
+        let mut result = Vec::new();
+        loop {
+            match args_it.next() {
+                None => break result,
+                Some("--") => {
+                    result.extend(args_it);
+                    break result;
+                }
+                Some(arg) if arg_follows_space.contains(arg) => {
+                    let _skip_parameter = args_it.next();
+                }
+                Some(arg) if !arg.starts_with('-') => {
+                    result.push(arg);
+                }
+                Some(_) => { /* skip: --these -and --also=this */ }
+            }
+        }
+    }
+
+    pub fn guess_git_blame_filename_extension(args: &[String]) -> ProcessArgs<String> {
+        {
+            let mut it = args.iter();
+            match (it.next(), it.next()) {
+                // git blame or git -C/-c etc. and then (maybe) blame
+                (Some(git), Some(blame))
+                    if git.contains("git") && (blame == "blame" || blame.starts_with('-')) => {}
```

Comment:
What's the thinking behind `contains` rather than `==`?

Reply:
```rust
(blame == "blame" || blame.starts_with('-'))
```

I was failing to understand this function yesterday -- if `blame.starts_with('-')` then when do we get round to checking that the string `blame` does occur at all?

Reply:
Yes, to match `git.exe` and other paths, but this would also pick up `magit` but fail on `Git.exe` - something `Path` based and removing the extension would be better. In the current patch however I completely ignore it (`_git`).

Regarding `blame`, I now parse everything, so it should always be present in the second position.